nibabel/nicom/utils.py

```diff
@@ -27,26 +27,20 @@ def find_private_section(dcm_data, group_no, creator):
     Returns
     -------
     element_start : int
-        Element number at which named section starts
+        Element number at which named section starts.
     """
-    is_regex = hasattr(creator, 'search')
-    if not is_regex:  # assume string / bytes
+    if hasattr(creator, 'search'):
+        match_func = lambda x : creator.search(x)
+    else:  # assume string / bytes
         creator = asstr(creator)
-    for element in dcm_data:  # Assumed ordered by tag (groupno, elno)
-        grpno, elno = element.tag.group, element.tag.elem
-        if grpno > group_no:
-            break
-        if grpno != group_no:
-            continue
+        match_func = lambda x : x == creator
```

Comment:
Likewise, we could use `creator.__eq__`, but I wouldn't insist on that if you prefer the readability of the `lambda`. Either way, I think it would be cleaner to include the `asstr` in this line. Either:

```suggestion
        match_func = lambda x: x == asstr(creator)
```

Or:

```suggestion
        match_func = asstr(creator).__eq__
```

Reply:
It has the annoying feature that we needlessly call `asstr` on `creator` for each element we test.

Reply:
Fair enough.

Reply:
Sorry - no - I'm wrong for the second - that's fine - the conversion is done once.